strum_macros/src/enum_iter.rs

```diff
@@ -79,9 +89,7 @@ pub fn enum_iter_inner(ast: &syn::DeriveInput) -> TokenStream {
             type Item = #name #ty_generics;
 
             fn next(&mut self) -> Option<#name #ty_generics> {
-                let output = match self.idx {
-                    #(#arms),*
-                };
+                let output = self.get(self.idx);
```

Comment:
We actually need to add another check here. Iterating forward and backward shouldn't overlap so if `idx` is greater than `back_idx` we should return `None`